app/components/e-textarea.vue

```diff
@@ -0,0 +1,277 @@
+<template>
+  <div :class="b(modifiers)">
+    <textarea :class="b('field')"
+              :name="name"
+              :disabled="disabled"
+              v-bind="$attrs"
+              @focus="onFocus"
+              @blur="onBlur"
+              @input="onInput">
+    </textarea>
+    <span v-if="!hasDefaultState && !hasFocus" :class="b('icon-splitter')"></span>
+    <div v-if="notification && hasFocus" :class="b('notification')">
+      <c-form-notification :state="state" v-html="notification"/>
+    </div>
+  </div>
+</template>
+
+<script>
+  import formStates from '@/mixins/form-states';
+  import CFormNotification from '@/components/c-form-notification';
+
+  export default {
+    name: 'e-textarea',
+    components: {
+      CFormNotification
+    },
+    mixins: [formStates],
+    inheritAttrs: false,
+
+    props: {
+      /**
+       * Value passed by v-model
+       */
+      value: {
+        default: null,
+        type: String,
+      },
+
+      /**
+       * Adds name attribute
+       */
+      name: {
+        required: true,
+        type: String,
+      },
+
+      /**
+       * Adds a notification block in a state container below the textarea
+       */
+      notification: {
+        type: String,
+        default: null
+      }
+    },
+    // data() {
+    //   return {};
+    // },
+
+    computed: {
+
+      /**
+       * Defines state modifier classes
+       *
+       * @returns  {Object}   BEM classes
+       */
+      modifiers() {
+        return {
+          ...this.stateModifiers,
+          notification: Boolean(this.$props.notification && this.hasFocus)
```

Comment:
- [ ] `this.$props.notification && this.hasFocus` should always be a boolean not? So "Boolean(...)" is not necessary

Reply:
I thought so as well, but It could actually return undefined if either both variables are undefined or if one is undefined and the other one is true.
See: https://jsfiddle.net/ycy3ymmp/